src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -1643,41 +1652,32 @@ namespace PoweredInductionUnits {
 		// Set the mass flow rates
 		if ( UnitOn ) {
 			// unit is on
-			// Special controls if FanOnFlowFrac = 0.0
+			// Calculate if reheat is needed
 			bool ReheatRequired = false;
-			if (PIU(PIUNum).FanOnFlowFrac <= 0.0) {
-				// Calculate if reheat is needed
-				Real64 qMinPrimary = PriAirMassFlowMin * ( CpAirZn * min( -SmallTempDiff, ( Node( PriNode ).Temp - Node( ZoneNode ).Temp ) ) );
-				if (  qMinPrimary < QToHeatSetPt ) ReheatRequired = true;
-			}
+			Real64 qMinPrimary = PriAirMassFlowMin * ( CpAirZn * min( -SmallTempDiff, ( Node( PriNode ).Temp - Node( ZoneNode ).Temp ) ) );
+			if (  qMinPrimary < QToHeatSetPt ) ReheatRequired = true;
 
 			if ( ! PriOn ) {
 				// no primary air flow
 				PriAirMassFlow = 0.0;
-				// PIU fan off if FanOnFlowFrac is 0.0 and there is no heating load, also reset fan flag if fan should be off
-				if ( PIU( PIUNum ).FanOnFlowFrac <= 0.0 ) {
-					if ( QZnReq <= SmallLoad ) {
-						SecAirMassFlow = 0.0;
-						PIUTurnFansOn = false;
-					} else {
-						SecAirMassFlow = PIU( PIUNum ).MaxSecAirMassFlow;
-						PIUTurnFansOn = ( DataHVACGlobals::TurnFansOn || DataHVACGlobals::TurnZoneFansOnlyOn );
-					}
+				// PIU fan off if there is no heating load, also reset fan flag if fan should be off
+				if ( QZnReq <= SmallLoad ) {
+					SecAirMassFlow = 0.0;
+					PIUTurnFansOn = false;
 				} else {
 					SecAirMassFlow = PIU( PIUNum ).MaxSecAirMassFlow;
+					PIUTurnFansOn = ( DataHVACGlobals::TurnFansOn || DataHVACGlobals::TurnZoneFansOnlyOn );
 				}
 			} else if ( CurDeadBandOrSetback( ZoneNum ) || std::abs( QZnReq ) < SmallLoad ) {
 				// in deadband or very small load: set primary air flow to the minimum
 				PriAirMassFlow = PriAirMassFlowMin;
-				// PIU fan off if FanOnFlowFrac is 0.0 and reheat is not needed, also reset fan flag if fan should be off
-				if ( ( PIU( PIUNum ).FanOnFlowFrac <= 0.0 ) && ReheatRequired ) {
+				// PIU fan off if reheat is not needed, also reset fan flag if fan should be off
+				if ( ReheatRequired ) {
```

Comment:
Now there is no decision, the choice was made prior to here. Just turn on if needed or turn off.